src/client/metadataTransfer.ts

```diff
@@ -216,6 +221,28 @@ export abstract class MetadataTransfer<Status extends MetadataRequestStatus, Res
     return { completed, payload: mdapiStatus as unknown as AnyJson };
   }
 
+  /**
+   * Based on the source components in the component set, it will return a polling frequency in milliseconds
+   */
+  private calculatePollingFrequency(): number {
+    const size = this.components?.getSourceComponents().toArray().length || 0;
+    // take a piece-wise approach to encapsulate discrete deployment sizes in polling frequencies that "feel" good when deployed
+    if (size === 0) {
+      // no component set size is possible for retrieve
```

Comment:
What's the thought behind having a large polling frequency here? I'm not quite clear on the comment.

Reply:
there is no way to calculate the size of SourceComponents in a retrieve request, in those cases, the `size` will be 0, and so to avoid polling excessively, we take a cautious approach and default to 1000ms

Reply:
got it, thanks!